src/LiveExampleSourcecode.vue

```diff
@@ -0,0 +1,137 @@
+<script setup lang="ts">
+import { type Ref, type PropType, ref, computed, onMounted, watch, nextTick } from "vue";
+import { type ExpandAnimation, expandAnimation } from "./expand-animation";
+import { generateId, getSourceCode } from "./utils";
+
+const props = defineProps({
+    element: {
+        type: HTMLElement,
+        required: true,
+    },
+    template: {
+        type: String,
+        required: true,
+    },
+    templateLanguage: {
+        type: String as PropType<"vue" | "html">,
+        required: true,
+    },
+});
+
+let idPrefix = generateId(props.template);
+const sourcecode = ref("");
+const expandable = ref(null) as Ref<unknown> as Ref<HTMLElement>;
```

Comment:
Is it possible to avoid `as` casting?

```suggestion
// inferred type: Ref<HTMLElement | undefined>
const expandable = ref<HTMLElement>();
```
or 
```suggestion
const expandable = ref<HTMLElement | null>(null);
```